### PR TITLE
[POI] add action handling code 

### PR DESCRIPTION
**Pull request by xieofxie**
### Purpose
*What is the context of this pull request? Why is it being done?*

2nd step of #2995

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

* add RouteType for validation since PVA doesn't support enum in json
* add DestinationActionType with logic for ShowRoute attribute
* add action handling code 
* add action Tests

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

File: skills/csharp/pointofinterestskill/Dialogs/MainDialog.cs [hunk 1]
```diff
@@ -282,6 +283,41 @@ private async Task<DialogTurnResult> RouteStepAsync(WaterfallStepContext stepCon
...
                 // Handle skill action logic here
+                var ev = activity.AsEventActivity();
+
+                if (!string.IsNullOrEmpty(ev.Name))
+                {
+                    switch (ev.Name)
+                    {
+                        case "GetDirectionAction":
+                            {
+                                await DigestActoinInput<GetDirectionInput>(stepContext, ev);
```
Comment (darrenj) [inline skills/csharp/pointofinterestskill/Dialogs/MainDialog.cs]: Typo - Actioi should be Action

File: skills/csharp/pointofinterestskill/Dialogs/PointOfInterestDialogBase.cs [hunk 2]
```diff
@@ -477,6 +478,29 @@ protected async Task<DialogTurnResult> ProcessPointOfInterestSelection(Waterfall
                         options.Prompt = ResponseManager.GetCardResponse(promptResponse, card, null);
                     }
 
+                    if (state.DestinationActionType != DestinationActionType.None)
```
Comment (darrenj) [inline skills/csharp/pointofinterestskill/Dialogs/PointOfInterestDialogBase.cs:482]: Code comments? What is this code doing?

File: skills/csharp/pointofinterestskill/wwwroot/manifest/manifest-1.0.json [hunk 3]
```diff
@@ -124,7 +124,7 @@
         },
         "showRoute": {
           "type": "boolean",
-          "description": "Whether show route."
+          "description": "If provided, determine whether show route without prompt."
```
Comment (darrenj) [inline skills/csharp/pointofinterestskill/wwwroot/manifest/manifest-1.0.json]: Indicates whether to show the route without a prompt

File: skills/csharp/pointofinterestskill/wwwroot/manifest/manifest-1.0.json [hunk 4]
```diff
@@ -157,7 +157,7 @@
...
```
Comment (darrenj) [inline skills/csharp/pointofinterestskill/wwwroot/manifest/manifest-1.0.json]: Indicates whether to show the route without a prompt

File: skills/csharp/pointofinterestskill/wwwroot/manifest/manifest-1.1.json [hunk 5]
```diff
@@ -188,7 +188,7 @@
         },
         "showRoute": {
           "type": "boolean",
-          "description": "Whether show route."
+          "description": "If provided, determine whether show route without prompt."
```
Comment (darrenj) [inline skills/csharp/pointofinterestskill/wwwroot/manifest/manifest-1.1.json]: Indicates whether to show the route without a prompt

File: skills/csharp/pointofinterestskill/wwwroot/manifest/manifest-1.1.json [hunk 6]
```diff
@@ -221,7 +221,7 @@
...
```
Comment (darrenj) [inline skills/csharp/pointofinterestskill/wwwroot/manifest/manifest-1.1.json]: Indicates whether to show the route without a prompt

File: skills/csharp/tests/pointofinterestskill.tests/Flow/Utterances/FindParkingUtterances.cs [hunk 7]
```diff
@@ -10,17 +12,31 @@ namespace PointOfInterestSkill.Tests.Flow.Utterances
 {
     public class FindParkingUtterances : BaseTestUtterances
     {
+        public static readonly string FindParkingNearby = "find a parking garage";
+
+        public static readonly string FindParkingNearest = "find a nearest parking garage";
```
Comment (darrenj) [inline skills/csharp/tests/pointofinterestskill.tests/Flow/Utterances/FindParkingUtterances.cs]: find the nearest parking garage

File: skills/csharp/tests/pointofinterestskill.tests/Flow/Utterances/FindPointOfInterestUtterances.cs [hunk 8]
```diff
@@ -10,20 +12,34 @@ namespace PointOfInterestSkill.Tests.Flow.Utterances
 {
     public class FindPointOfInterestUtterances : BaseTestUtterances
     {
+        public static readonly string WhatsNearby = "What's nearby?";
+
+        public static readonly string FindNearestPoi = "find closest poi nearby me";
```
Comment (darrenj) [inline skills/csharp/tests/pointofinterestskill.tests/Flow/Utterances/FindPointOfInterestUtterances.cs]: near me

File: skills/csharp/tests/pointofinterestskill.tests/Flow/Utterances/FindPointOfInterestUtterances.cs [hunk 9]
```diff
@@ -10,20 +12,34 @@ namespace PointOfInterestSkill.Tests.Flow.Utterances
...
+
+        public static readonly string FindPharmacy = $"find a {ContextStrings.Pharmacy} nearby me";
```
Comment (darrenj) [inline skills/csharp/tests/pointofinterestskill.tests/Flow/Utterances/FindPointOfInterestUtterances.cs]: nearby, remove the me

File: skills/csharp/tests/pointofinterestskill.tests/Flow/Utterances/RouteFromXToYUtterances.cs [hunk 10]
```diff
@@ -11,20 +13,32 @@ namespace PointOfInterestSkill.Tests.Flow.Utterances
 {
     public class RouteFromXToYUtterances : BaseTestUtterances
     {
+        public static readonly string FindRoute = "find a route";
+
+        public static readonly string GetToMicrosoft = $"get directions to {ContextStrings.MicrosoftCorporation}";
+
+        public static readonly string GetToMicrosoftNearAddress = $"get directions to {ContextStrings.MicrosoftCorporation} near {ContextStrings.Ave}";
+
+        public static readonly string GetToNearestPharmacy = $"get directions the nearest {ContextStrings.Pharmacy}";
```
Comment (darrenj) [inline skills/csharp/tests/pointofinterestskill.tests/Flow/Utterances/RouteFromXToYUtterances.cs]: get directions to the nearest

**Review by darrenj — CHANGES_REQUESTED** [review]

File: skills/csharp/pointofinterestskill/Dialogs/MainDialog.cs (hunk 1 above)
Comment (xieofxie) [reply skills/csharp/pointofinterestskill/Dialogs/MainDialog.cs]: 😱

File: skills/csharp/pointofinterestskill/Dialogs/PointOfInterestDialogBase.cs (hunk 2 above)
Comment (xieofxie) [reply skills/csharp/pointofinterestskill/Dialogs/PointOfInterestDialogBase.cs:482]: Added

**Review by lzc850612 — APPROVED** [review]

**Pull request merged**